specifyweb/frontend/js_src/lib/components/FormEditor/Create.tsx

```diff
@@ -112,35 +112,56 @@ function ListViews({
   readonly views: AllTableViews['database'] | AllTableViews['disk'];
   readonly onSelect: (view: ViewDefinition) => void;
 }): JSX.Element {
-  const grouped = React.useMemo(
-    () => group(views.map((view) => [view.category, view] as const)),
-    [views]
-  );
+  const grouped = React.useMemo(() => {
+    const byDiscipline = group(
+      views.map((view) => [view.disciplineId, view] as const)
+    );
+    return byDiscipline.map(
+      ([parentKey, children]) =>
+        [
+          parentKey,
+          group(children.map((view) => [view.category, view])),
+        ] as const
+    );
+  }, [views]);
+
   const [preview, setPreview] = React.useState<ViewDefinition | undefined>(
     undefined
   );
+
+  const { disciplines } = useOutletContext<FormEditorOutlet>();
+
   return (
     <div className="flex flex-col gap-4">
       <h3 className={className.headerPrimary}>{header}</h3>
       <div className="flex flex-col gap-2">
-        {grouped.map(([category, views], index) => (
+        {grouped.map(([disciplineId, categories], index) => (
           <div className="flex flex-col gap-2" key={index}>
-            <h4 className={className.headerGray}>{category}</h4>
+            <h4 className={`${className.headerGray} font-bold`}>
+              {disciplines.find(({ id }) => id === disciplineId)?.name}
+            </h4>
             <Ul className="flex flex-col gap-2">
-              {views.map((view, index) => (
-                <li className="flex gap-2" key={index}>
-                  <Button.LikeLink onClick={(): void => setPreview(view)}>
-                    {view.name}
-                  </Button.LikeLink>
-                  {typeof view.editUrl === 'string' && (
-                    <Link.Icon
-                      className={className.dataEntryEdit}
-                      href={view.editUrl}
-                      icon="pencil"
-                      title={commonText.edit()}
-                    />
-                  )}
-                </li>
+              {categories.map(([category, views], index) => (
+                <div className="flex flex-col gap-2" key={index}>
+                  <h4 className={className.headerGray}>{category}</h4>
```

Comment:
now that there are so many nested levels, semantically it might be better to represent this using nested ul and li, rather than div+h3,4,5

i.e you have nested lists, not sections with subsections, thus ul+li makes more sense than headers